File: tests/cpp/test_doom_hud.cpp
// examples/doom/hud.md 검증 (C++17).
#include "doom_hud.hpp"
#include "minitest.hpp"

using namespace Doom;

TEST(hud_low_health_shows_hurt_face) {
    auto actor = Actor::Spawn(monsterTypes().at("Zombieman"), 0, 0);
    Player player(actor);
    player.health = 10;
    Hud hud;
    hud.updateFace(player);
    CHECK(hud.faceFrame.find("STFEVL") != std::string::npos);
}

TEST(hud_full_health_shows_calm_face) {
    auto actor = Actor::Spawn(monsterTypes().at("Zombieman"), 0, 0);
    Player player(actor);
    player.health = 100;
    Hud hud;
    hud.updateFace(player);
    CHECK(hud.faceFrame.find("STFST") != std::string::npos);
}

TEST(hud_zero_health_shows_dead_face) {
    auto actor = Actor::Spawn(monsterTypes().at("Zombieman"), 0, 0);
    Player player(actor);
    player.health = 0;
    Hud hud;
    hud.updateFace(player);
    CHECK_EQ(hud.faceFrame, std::string("STFDEAD0"));
}

TEST(hud_recent_hit_shows_directional_face) {
    auto actor = Actor::Spawn(monsterTypes().at("Zombieman"), 0, 0);
    Player player(actor);
    player.health = 80;
    Hud hud;
    hud.updateFace(player, std::optional<std::string>("left"));
    CHECK(hud.faceFrame.find("HIT_left") != std::string::npos);
}

TEST_MAIN()

File: tests/test_doom_hud.py
import unittest

from doom_actors import Actor, MONSTER_TYPES, Player
from doom_hud import Hud


class HudFaceExampleTest(unittest.TestCase):
    """hud.md의 '## 예제: 체력이 0에 가까우면 고통스러운 얼굴'을 그대로 옮긴 테스트."""

    def test_low_health_shows_hurt_face(self):
        actor = Actor.Spawn(MONSTER_TYPES["Zombieman"], 0, 0)
        player = Player(actor)
        player.health = 10
        hud = Hud()
        hud.updateFace(player)
        self.assertIn("STFEVL", hud.faceFrame)


class HudFaceGradientTest(unittest.TestCase):
    def test_full_health_shows_calm_face(self):
        actor = Actor.Spawn(MONSTER_TYPES["Zombieman"], 0, 0)
        player = Player(actor)
        player.health = 100
        hud = Hud()
        hud.updateFace(player)
        self.assertIn("STFST", hud.faceFrame)

    def test_zero_health_shows_dead_face(self):
        actor = Actor.Spawn(MONSTER_TYPES["Zombieman"], 0, 0)
        player = Player(actor)
        player.health = 0
        hud = Hud()
        hud.updateFace(player)
        self.assertEqual(hud.faceFrame, "STFDEAD0")

    def test_recent_hit_shows_directional_face(self):
        actor = Actor.Spawn(MONSTER_TYPES["Zombieman"], 0, 0)
        player = Player(actor)
        player.health = 80
        hud = Hud()
        hud.updateFace(player, justDamagedFromDirection="left")
        self.assertIn("HIT_left", hud.faceFrame)


if __name__ == "__main__":
    unittest.main()
